asalSayi.java

```diff
@@ -0,0 +1,55 @@
+package asal;
+import java.util.*;
+
+
+public class asalSayi {
+	
+	public static void main(String args[])
+	{	
+		short sayi;
+		int x=0;
+		Scanner scanner=new Scanner(System.in);
+		System.out.println("Sayi giriniz");
+		
+		if(scanner.hasNextShort())
+		{
+			sayi=scanner.nextShort();
+			
+			if(sayi>0)
+			{
+				for(int i=2;i<sayi/2;i++)
```

Comment:
Bu satırda i'nin nihai değeri "i<sayi" değil "i<=sayi" olmalı. Aksi taktirde 4 sayısı için sonuç asal çıkıyor.